### PR TITLE
OpenH264 upgrade to v2.1.1 via conda-forge

### PR DESCRIPTION
**Pull request by tobijk**
This project broke ABI from 2.0.0 to 2.1.0, but they forgot to bump the soversion. They bumped the soversion from 5 to 6 in 2.1.1. But we have already published 2.1.0.

This merges the conda-forge recipe for 2.1.1 into our master (which is at 1.8.0) **including the changes of the soversion**. Other than that there are no actual functional changes between 2.1.0 and 2.1.1.

Additionally changed:

* only require nasm on x86 (it's an x86x-only assembler)
* ignore_run_exports: zlib

Current situation:
```
--------------------------------------------------------------------
# linux-64
--------------------------------------------------------------------
* openh264                       1.8.0      hd408876_0  pkgs/main
* openh264                       2.1.0      hd408876_0  pkgs/main

--------------------------------------------------------------------
# linux-ppc64le
--------------------------------------------------------------------
* openh264                       2.1.0      hd408876_0  pkgs/main

--------------------------------------------------------------------
# linux-aarch64
--------------------------------------------------------------------
* openh264                       1.8.0      h22f4aa5_0  pkgs/main

--------------------------------------------------------------------
# linux-s390x
--------------------------------------------------------------------
->Package "openh264" not found!

--------------------------------------------------------------------
# osx-64
--------------------------------------------------------------------
* openh264                       1.8.0      hd9629dc_0  pkgs/main
* openh264                       2.1.0      hd9629dc_0  pkgs/main

--------------------------------------------------------------------
# osx-arm64
--------------------------------------------------------------------
* openh264                       1.8.0      h98b2900_0  pkgs/main

--------------------------------------------------------------------
# win-32
--------------------------------------------------------------------
->Package "openh264" not found!

--------------------------------------------------------------------
# win-64
--------------------------------------------------------------------
->Package "openh264" not found!
```

The ABI was bumped because of [1] there are no functional changes from v2.1.0 [2]

[1] https://github.com/cisco/openh264/issues/3264
[2] https://github.com/cisco/openh264/compare/v2.1.0..v2.1.1

### CODE REVIEW

**Review by katietz — APPROVED** [review]

**Review by katietz — APPROVED** [review]

File: .ci_support/linux_aarch64_.yaml [hunk 1]
```diff
@@ -2,20 +2,29 @@ BUILD:
 - aarch64-conda_cos7-linux-gnu
 c_compiler:
 - gcc
+c_compiler_version:
+- '9'
```
Comment (katietz) [inline .ci_support/linux_aarch64_.yaml:6]: why those quotes?

File: .ci_support/linux_aarch64_.yaml [hunk 2]
```diff
@@ -2,20 +2,29 @@ BUILD:
...
 - cos7
 channel_sources:
-- conda-forge,c4aarch64,defaults
+- conda-forge
 channel_targets:
 - conda-forge main
 cxx_compiler:
 - gxx
+cxx_compiler_version:
+- '9'
```
Comment (katietz) [inline .ci_support/linux_aarch64_.yaml:18]: Likewise

File: .ci_support/linux_ppc64le_.yaml [hunk 3]
```diff
@@ -1,15 +1,26 @@
...
+cdt_name:
+- cos7
 channel_sources:
 - conda-forge,defaults
 channel_targets:
 - conda-forge main
 cxx_compiler:
 - gxx
+cxx_compiler_version:
+- '9'
```
Comment (katietz) [inline .ci_support/linux_ppc64le_.yaml:14]: again

File: .ci_support/linux_ppc64le_.yaml [hunk 4]
```diff
@@ -1,15 +1,26 @@
...
 pin_run_as_build:
   zlib:
     max_pin: x.x
+target_platform:
+- linux-ppc64le
+zip_keys:
+- - c_compiler_version
+  - cxx_compiler_version
 zlib:
 - '1.2'
```
Comment (katietz) [inline .ci_support/linux_ppc64le_.yaml:26]: ...

**Review by katietz — CHANGES_REQUESTED** [review]

**Review by varlackc — COMMENTED** [review]
The Dev_url and Doc_url seem to point to the same location.

https://github.com/AnacondaRecipes/openh264-feedstock/blob/update/merge-cf-2.1.1/recipe/meta.yaml#L46&&#L47

maybe we can point to a different location on the doc_url, it might be a good idea to look at the official website:

http://www.openh264.org/

File: .ci_support/linux_aarch64_.yaml (hunk 1 above)
Comment (tobijk) [reply .ci_support/linux_aarch64_.yaml:6]: conda-forge specific, closing as informative

File: .ci_support/linux_aarch64_.yaml (hunk 2 above)
Comment (tobijk) [reply .ci_support/linux_aarch64_.yaml:18]: conda-forge specific, closing as informative

File: .ci_support/linux_ppc64le_.yaml (hunk 3 above)
Comment (tobijk) [reply .ci_support/linux_ppc64le_.yaml:14]: conda-forge specific, closing as informative

File: .ci_support/linux_ppc64le_.yaml (hunk 4 above)
Comment (tobijk) [reply .ci_support/linux_ppc64le_.yaml:26]: conda-forge specific, closing as informative

**Comment by tobijk** [comment]
@varlackc, thanks for the suggestion, there seems to be Doxygen documentation inside the source tree, but unfortunately there isn't a pre-built version of it available anywhere on the homepage. So I think we should leave it as it is.

**Comment by tobijk** [comment]
Added compatibility symlinks to keep programs linked avainst soversion 5 running. In our repo that is only ffmpeg. I have tested that ffmpeg loads correctly with the symlink in place on linux-64 and osx-64. On some arches ffmpeg has been built against openh264 1.8.0, which uses soversion 4. However, since ffmpeg depends on "openh264 >=1.8.0,<1.9.0a0" in that case, I trust that conda's solver will do the right thing. We might consider to rebuilt ffmpeg later.

**Review by katietz — APPROVED** [review]
lgtm

**Review by varlackc — APPROVED** [review]

**Pull request merged**